client/src/game/shapes/asset.ts

```diff
@@ -27,7 +32,35 @@ export class Asset extends BaseRect {
     draw(ctx: CanvasRenderingContext2D): void {
         super.draw(ctx);
         try {
-            ctx.drawImage(this.img, g2lx(this.refPoint.x), g2ly(this.refPoint.y), g2lz(this.w), g2lz(this.h));
+
+            if (this.rotationDegrees == 0) {
+                ctx.drawImage(this.img, g2lx(this.refPoint.x), g2ly(this.refPoint.y), g2lz(this.w), g2lz(this.h));
+            }
+            else {
+
+                ctx.save();
+
+                // translate from the center of the image
+                ctx.translate((g2lx(this.refPoint.x) + (g2lz(this.w) / 2)), (g2ly(this.refPoint.y) + (g2lz(this.h) / 2)))
+
+                // rotate the canvas to the specified degrees
+                ctx.rotate(this.rotationDegrees * Math.PI / 180);
+
+                // find the new 0,0 point
+                if (this.rotationDegrees == 90 || this.rotationDegrees == 270)
+                    ctx.translate(-(g2ly(this.refPoint.y) + (g2lz(this.h) / 2)), -(g2lx(this.refPoint.x) + (g2lz(this.w) / 2)))
+                else
+                    ctx.translate(-(g2lx(this.refPoint.x) + (g2lz(this.w) / 2)), -(g2ly(this.refPoint.y) + (g2lz(this.h) / 2)))
+
+                // draw the image
+                if (this.rotationDegrees == 90 || this.rotationDegrees == 270)
+                    ctx.drawImage(this.img, g2ly(this.refPoint.y), g2lx(this.refPoint.x), g2lz(this.h), g2lz(this.w));
+                else
+                    ctx.drawImage(this.img, g2lx(this.refPoint.x), g2ly(this.refPoint.y), g2lz(this.w), g2lz(this.h));
+
+                // we�re done with the rotating so restore the unrotated context
+                ctx.restore();
```

Comment:
Although I'm sure this works, I think it's faster to use something like this:
```javascript
ctx.setTransform(1, 0, 0, 1, x, y);
ctx.rotate(rotation); // in radians
ctx.drawImage(image, -image.width / 2, -image.height / 2);
ctx.setTransform(1,0,0,1,0,0); // reset the transform
```
As it does not require context saving/restoring.  What do you think ?

Reply:
I played with a few methods, and started down a few paths until I hit the selection code, which lead to this solution. Mainly so I didn't have to overhaul selection and resizing to deal with a rotated Asset.

I haven't done much with canvas so this might not be the best solution (but it works).

Rotating from the center point doesn't quite work. Snap to grid works fine with even number of tiles, but on odd number of tiles it leaves the image offset by half a tile.

Not sure I see what your doing until I run it in code. Will review.